src/parcsr_ls/par_cycle.c

```diff
@@ -286,6 +286,15 @@ hypre_BoomerAMGCycle( void              *amg_vdata,
    hypre_GpuProfilingPushRange(nvtx_name);
    while (Not_Finished)
    {
+#ifdef HYPRE_USING_NODE_AWARE_MPI
+      if (level >= hypre_HandleNodeAwareSwitchoverThreshold(hypre_handle()))
```

Comment:
@liruipeng This switch threshold could be moved inside the AMG struct

Reply:
OK. We have the access to `hypre_ParAMGData` in this function. Do we need `hypre_HandleNodeAwareSwitchoverThreshold` in other functions that don't have the access?

Reply:
I can only see it being used at `par_cycle.c`

Reply:
So I didn't check these in as permanent changes to the ij driver but to properly use node aware, you must set this hypre_HandleNodeAwareSwitchoverThreshold otherwise it will unoptimally use node aware mpi for every amg level. 

This brings up 2 things:
1. `NodeAwareSwitchoverThreshold` must be accessible to set like from the ij.c driver. It's quite nice to have it as a command line arg to test different thresholds without recompiling or anything.

2. We probably want to document this and how to connect to MPI advance. I can compile the steps and maybe we could include it somewhere in the wiki or something? But I don't know how to contribute to the wiki.

Reply:
1. Yes, there should be a Set routine for that, e.g., `HYPRE_BoomerAMGSetNodeAwareSwitchLevel`
2. Great idea, I'm not sure about the access but if you prepare something in markdown format, we can add it to the wiki.

Reply:
Document it in the Wiki is a good idea.